src/N3Store.js

```diff
@@ -790,9 +820,139 @@ export default class N3Store {
       this.removeQuads(toRemove);
     return lists;
   }
+
+  // ### Store is an iterable.
+  // Can be used where iterables are expected: for...of loops, array spread operator,
+  // `yield*`, and destructuring assignment (order is not guaranteed).
+  *[Symbol.iterator]() {
+    yield* this.getQuads();
+  }
 }
 
 // Determines whether the argument is a string
 function isString(s) {
   return typeof s === 'string' || s instanceof String;
 }
+
+/**
+ * A class that implements both the DatasetCore interface and the Readable stream
+ * interface.
+ */
+class DatasetCoreAndReadableStream {
```

Comment:
I'm open to suggestions to improve the implementation. I believe I captured all possible public getters and methods of the Readable stream interface.